### PR TITLE
Merge first and last names into name

### PR DESCRIPTION
**Pull request by jferris**
GitHub auth (among other providers) only provides a "name" and not distinct
first and last names. We only ever use the full name, so we can safely merge
these into a single name when we authenticate into providers which provide
both.

### CODE REVIEW

File: Foundation.hs [hunk 1]
```diff
@@ -149,12 +151,19 @@ instance YesodAuth App where
...
 buildUser (Creds csPlugin csIdent csExtra) =
-    User <$> lookup "first_name" csExtra
-         <*> lookup "last_name" csExtra
+    User <$> buildName csExtra
          <*> lookup "email" csExtra
          <*> pure csPlugin
          <*> pure csIdent
 
+buildName :: [(Text, Text)] -> Maybe Text
+buildName csExtra = T.intercalate " " <$> parts
```
Comment (jferris) [inline Foundation.hs]: I think maybe this is more readable with `do` syntax:

``` haskell
buildName :: [(Text, Text)] -> Maybe Text
buildName csExtra = do
    firstName <- lookup "first_name" csExtra
    lastName <- lookup "last_name" csExtra
    return $ firstName <> " " <> lastName
```

Thoughts?

File: migrations/merge-user-first-and-last-name-into-name.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE "user" ADD COLUMN name varchar;
+UPDATE "user" SET name = first_name || ' ' || last_name;
```
Comment (pbrisbin) [inline migrations/merge-user-first-and-last-name-into-name.sql]: `||` is string concatenation? Weird.

File: tests/TestHelpers/DB.hs [hunk 3]
```diff
@@ -36,8 +36,7 @@ runDB query = do
 createUser :: Text -> Example (Entity User)
 createUser ident = do
     insertEntity User
-        { userFirstName = "John" <> ident
-        , userLastName  = "Smith"
+        { userName      = "John " <> ident <> " Smith"
```
Comment (pbrisbin) [inline tests/TestHelpers/DB.hs]: Does it break anything to do `"John Smith (" <> ident <> ")"`? I think that'd be better overall, but I understand if you want to keep the changes here to a minimum.

File: Foundation.hs [hunk 4]
```diff
@@ -149,12 +151,17 @@ instance YesodAuth App where
...
+    User <$> buildName csExtra
          <*> lookup "email" csExtra
          <*> pure csPlugin
          <*> pure csIdent
 
+buildName :: [(Text, Text)] -> Maybe Text
+buildName csExtra = do
+    firstName <- lookup "first_name" csExtra
+    lastName <- lookup "last_name" csExtra
+    return $ firstName <> " " <> lastName
```
Comment (pbrisbin) [inline Foundation.hs]: Thoughts on moving `buildUser`, `buildName`, and `repalceUser` into `Model.User`?

**Comment by pbrisbin** [comment]
Minor comments -- LGTM

File: migrations/merge-user-first-and-last-name-into-name.sql (hunk 2 above)
Comment (jferris) [inline migrations/merge-user-first-and-last-name-into-name.sql]: It is string concatenation, and it is weird.

File: tests/TestHelpers/DB.hs (hunk 3 above)
Comment (jferris) [inline tests/TestHelpers/DB.hs]: It's not much different to do it in the parens way, and I don't really have a preference. I'll update it to use that style.

File: Foundation.hs (hunk 4 above)
Comment (jferris) [inline Foundation.hs]: That seems logical to me, but I think it means a larger restructuring than we can perform here:

```
Module imports form a cycle:
         module ‘Import’ (Import.hs)
        imports ‘Foundation’ (Foundation.hs)
  which imports ‘Model.User’ (./Model/User.hs)
  which imports ‘Import’ (Import.hs)
```

Comment (pbrisbin) [inline Foundation.hs]: Oh right. Only Model can be imported into Foundation. Ignore me.

**Pull request merged**